plugins/lookup/passwordstore.py

```diff
@@ -209,10 +212,15 @@ def check_pass(self):
             ).splitlines()
             self.password = self.passoutput[0]
             self.passdict = {}
-            for line in self.passoutput[1:]:
-                if ':' in line:
-                    name, value = line.split(':', 1)
-                    self.passdict[name.strip()] = value.strip()
+            try:
+                values = yaml.safe_load('\n'.join(self.passoutput[1:]))
+                for key, item in values.items():
+                    self.passdict[key] = item
+            except yaml.YAMLError:
```

Comment:
Do you know in which cases the result is not a YAML file? Also, how is the first line special?

Reply:
The default (even though just convention, not enforced) for pass is that the first line is simply the password for the service/host/whatever is stored there, with no additional metadata.

This is reflected by `pass -c <somepassword>` copying only the first line by default.

So in YAML it wouldn't be valid, which is why it isn't considered for YAML parsing.

Reply:
Ah, so it's basically one line with a specific formatting, followed by arbitrary text - but the module decided on a "most common format" and tries to parse that. And with this PR, it will also accept YAML and JSON (since JSON is a subset of YAML). :+1: